### ISSUE TITLE
Extract client to standalone module?

### ISSUE THREAD

**Issue by fillup** [nutanix/terraform-provider-nutanix#155]
I'm working on a project that integrates with Nutanix and I'd like to use the client you've already developed here, but it feels weird to import the Terraform module just to get access to the client package. Also being bundled in the Terraform module concerns me that it could be modified over time in ways that would be convenient for Terraform usage but make it less useful outside the Terraform context. 

Is there any possibility it could be moved out and then just imported into the Terraform module?

**Comment by PacoDw** [comment]
yes, you are right and I agree with that, we will work on it, thank you so much for your review

**Comment by fillup** [comment]
Awesome @PacoDw that would be fantastic! I'm just a few weeks away in my project from wanting to begin using the client. I assume that's a bit soon for you to be able to refactor this but it's great to know it will hopefully happen :-) Do you have any sense of how long it might take to do? I understand that's not an easy question to answer though.

**Comment by PacoDw** [comment]
In a quick estimated I think it would be done in two months :)

**Comment by thirdeyenick** [comment]
Hi there,
is there any news on this issue? We also would like to use the 'official' client here, but have the same concerns.

**Comment by Bermos** [comment]
Hello, sorry to resurrect such an old issue but since there is now also this client implementation https://github.com/nutanix-cloud-native/prism-go-client, are there any plans to use it for the terraform module as well? Seems a bit like a waste to maintain the client twice.
